maintainers/scripts/update-luarocks-packages

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -p nix-prefetch-scripts luarocks-nix -i bash
+
+# You'll likely want to use
+# ``
+# nixpkgs $ maintainers/scripts/update-luarocks-packages.sh > pkgs/development/lua-modules/generated-packages.nix
+# ``
+# to update all libraries in that folder.
+# to debug, redirect stderr to stdout with 2>&1
+
+
+# stop the script upon C-C
+set -eu -o pipefail
+
+if [ $# -lt 1 ]; then
+    echo "Usage: $0 GENERATED_FILENAME"
+    echo "(most likely pkgs/development/lua-modules/generated-packages.nix)"
+    exit 1
+fi
+
+GENERATED_NIXFILE="$1"
+PACKAGE=""
+# we have to use a temp file else
+TMP_FILE="$(mktemp)"
+
+exit_trap()
+{
+  local lc="$BASH_COMMAND" rc=$?
+  test $rc -eq 0 || echo "*** error $rc: $lc"
+}
+
+trap exit_trap EXIT
+
+
+# -p <PKG> package
+while getopts "p:" opt; do
+  case $opt in
+    p)
+      echo "Generating nix derivation for $OPTARG !" >&2
+      PACKAGE="$OPTARG"
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      ;;
+  esac
+done
+
+# read returns a non-zero return code when it reaches EOF
+# so disable the checks just for the 2 read
```

Comment:
1. Given you don't even use `"` you could as well do a normal variable assignment
2. I would consider using `|| true` with a comment instead of `set +e` — less chances of it getting its own life separate from the initial statements.

Reply:
```
HEADER="
/* ${GENERATED_NIXFILE} is an auto-generated file -- DO NOT EDIT!
Regenerate it with:
nixpkgs$ ${0} ${GENERATED_NIXFILE}

These packages are manually refined in lua-overrides.nix
*/
…
"

Reply:
I see. Done as you suggested. I removed the -e/+e that were helpful to diagnose weird read errors (zsh vs sh) that are not needed anymore.

Reply:
And maybe kill the comment?